cmd/proxy/main.go

```diff
@@ -227,25 +253,95 @@ func (p *Proxy) run(o *ProxyRunOptions) error {
 		return fmt.Errorf("failed to run the admin server: %v", err)
 	}
 
-	stopCh := make(chan struct{})
+	stopCh := SetupSignalHandler()
```

Comment:
Why not set it up the same way as in the client code? That seems cleaner.
```golang
ch := make(chan os.Signal, 1)
signal.Notify(ch)
go func() {
	<-ch
	klog.Info("Shutting down server.")
	if masterStop != nil {
		masterStop()
	}
        cancel()
}()

Reply:
Stealing from controller runtime. I think its a good idea for us to move in that direction.